R/supervised-model-deployment.R

```diff
@@ -89,16 +89,22 @@ SupervisedModelDeployment <- R6Class("SupervisedModelDeployment",
   self$params$debug <- p$debug
   
   if (!is.null(p$modelName))
-    self$params$modelName <- p$modelName
+  self$params$modelName <- p$modelName
 
   # for deploy method
   if (!is.null(p$cores))
   self$params$cores <- p$cores
+  
+  if (is.null(self$params$modelName))
+  self$params$modelName = private$modelName
   },
 
   loadData = function() {
-    cat('Loading Data...','\n')
+    # Load model info
+    cat('Loading Model Info...','\n')
+    private$loadModelAndInfo(private$modelName)
```

Comment:
That makes sense. I added a new private variable to all model classes that contains the model name. I changed the call to use that variable as the name and removed the redundant loading from deploy.